From 598211c2844a922156ccefeeb8ce98c44d4a0274 Mon Sep 17 00:00:00 2001
From: Stefan Riedmueller <s.riedmueller@phytec.de>
Date: Fri, 3 Dec 2021 16:17:36 +0100
Subject: [PATCH] waylandsink: Prefer ion allocator over dmabufheaps allocator

The ion memory allocator currently achieves a significantly better
performance over the dmabufheaps allocator. Thus use this one by
default.

Signed-off-by: Stefan Riedmueller <s.riedmueller@phytec.de>
---
 ext/wayland/gstwaylandsink.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/ext/wayland/gstwaylandsink.c b/ext/wayland/gstwaylandsink.c
index 9d4a9de2a475..08782ae7bd5d 100644
--- a/ext/wayland/gstwaylandsink.c
+++ b/ext/wayland/gstwaylandsink.c
@@ -644,12 +644,12 @@ gst_wayland_create_pool (GstWaylandSink * sink, GstCaps * caps)
   gst_video_info_from_caps (&info, caps);
   GstVideoFormat format = GST_VIDEO_INFO_FORMAT (&info);
   if (gst_wl_display_check_format_for_dmabuf (sink->display, format)) {
-#ifdef HAVE_DMABUFHEAPS_ALLOCATOR
-    alloc = gst_dmabufheaps_allocator_obtain ();
-#endif
 #ifdef HAVE_ION_ALLOCATOR
+    alloc = gst_ion_allocator_obtain ();
+#endif
+#ifdef HAVE_DMABUFHEAPS_ALLOCATOR
     if (!alloc)
-      alloc = gst_ion_allocator_obtain ();
+      alloc = gst_dmabufheaps_allocator_obtain ();
 #endif
   }
 
-- 
2.25.1

